thriftbp/thrifttest/server.go

```diff
@@ -113,11 +114,14 @@ func (s *Server) Start(ctx context.Context) {
 
 // Close the underying Server and Baseplate as well as the thriftbp.ClientPool.
 func (s *Server) Close() error {
-	closers := batchcloser.New(s.Server, s.Baseplate())
+	closers := make([]io.Closer, 0, 3)
+	// close the ClientPool first so the server doesn't hang waiting for it to
+	// close while trying to close itself.
 	if s.ClientPool != nil {
-		closers.Add(s.ClientPool)
+		closers = append(closers, s.ClientPool)
 	}
-	return closers.Close()
+	closers = append(closers, s.Server, s.Baseplate())
+	return batchcloser.New(closers...).Close()
```

Comment:
🔕 so the only change is the order of closing?  Why is this switching away from using batchcloser.New at the beginning?

Reply:
changed